src/ExerciseRunner/CgiRunner.php

```diff
@@ -43,6 +43,11 @@ class CgiRunner implements ExerciseRunnerInterface
      */
     public function __construct(CgiExercise $exercise, EventDispatcher $eventDispatcher)
     {
+        // Suppress shell output. Anything non-zero is a failure.
+        @system('php-cgi --version > /dev/null 2>&1', $failedToRun);
```

Comment:
Checking for the binary has to be a little more complex, as I found in some cases the binary wasn't in the path at all, and it just resided next to `php` which was in the path and also on Windows it needed `.exe` on the end (in my tests anyway).

See the code in the `getProcess()` method of the same check, something like:

```php
$cgi = sprintf('php-cgi%s', DIRECTORY_SEPARATOR === '\\' ? '.exe' : '');
$cgiBinary  = sprintf(
    '%s -dalways_populate_raw_post_data=-1 -dhtml_errors=0 -dexpose_php=0',
    realpath(sprintf('%s/%s', str_replace('\\', '/', dirname(PHP_BINARY)), $cgi))
);
```

Obviously though you don't need all those flags though.